### PR TITLE
Flatten (overload) via property of an object (issue #413)

### PR DESCRIPTION
**Pull request by leandromoh**
Closes issue #413 .Flatten (overload) via property of an object using a selector function (`Func<object, IEnumerable>`).

In a Later PR, we can add the overload  "predicate + selector sepatared", since we can create it in terms of this PR/MVP.

### CODE REVIEW

**Comment by leandromoh** [comment]
@atifaziz any chance to merge this one this week?

**Comment by leandromoh** [comment]
What is missing to publish It?

**Comment by atifaziz** [comment]
> What is missing to publish It?

See the [3.0 milestone](https://github.com/morelinq/MoreLINQ/milestone/2). The biggest item is #214, which is mostly done but needs some last polishing touches. It's been neglected because I have been busy reviewing PRs.

**Comment by leandromoh** [comment]
I see. Can we include Flatten Selector overload in 3.0?

**Comment by atifaziz** [comment]
> Will we include Flatten Selector overload in 3.0?

I'll try if I can find the time to review (unless @fsateler or someone else can beat me to it). It's not a breaking change so it can go into 3.1 soon after.

> This PR adds more about tests than implementation.

It adds new functionality too, no?

**Comment by leandromoh** [comment]
> It adds new functionality too, no?

yes, but in fact there is no more than 10 lines of implementation. Major part is tests.

**Comment by leandromoh** [comment]
@atifaziz how about this one?

**Comment by leandromoh** [comment]
@atifaziz?

**Comment by atifaziz** [comment]
@leandromoh Trying to enjoy the summer. 🌻

**Comment by leandromoh** [comment]
@atifaziz looks great! hope you enjoy!

**Comment by leandromoh** [comment]
any news ?

**Comment by atifaziz** [comment]
Next week, promise! 🤞

File: MoreLinq.Test/FlattenTest.cs [hunk 1]
```diff
@@ -289,5 +294,164 @@ public void FlattenEvaluatesInnerSequencesLazily()
...
+                    case Attribute a:
+                        return a.Values;
+                    default:
+                        return null;
+                }
+            });
+
+            var expectations = new object[] { "series1", 1, 2, 3, 4, "series2", 5, 6 };
+
+            Assert.That(result, Is.EquivalentTo(expectations));
```
Comment (atifaziz) [inline MoreLinq.Test/FlattenTest.cs]: `EquivalentTo` is not an _exact_ test of expectations. Per [docs](https://github.com/nunit/docs/wiki/CollectionEquivalentConstraint):

> **CollectionEquivalentConstraint** tests that two `IEnumerables` are equivalent - that they contain the same items, in any order.

```suggestion
            Assert.That(result, Is.EqualTo(expectations));
```

File: MoreLinq.Test/FlattenTest.cs [hunk 2]
```diff
@@ -289,5 +294,164 @@ public void FlattenEvaluatesInnerSequencesLazily()
...
+                    case IEnumerable inner:
+                        return inner;
+                    default:
+                        return Enumerable.Empty<object>();
+                }
+            });
+
+            var expectations = new object[] { 1, 2, 3, 4 };
+
+            Assert.That(result, Is.EquivalentTo(expectations));
```
Comment (atifaziz) [inline MoreLinq.Test/FlattenTest.cs]:
```suggestion
            Assert.That(result, Is.EqualTo(expectations));
```

File: MoreLinq.Test/FlattenTest.cs [hunk 3]
```diff
@@ -289,5 +294,164 @@ public void FlattenEvaluatesInnerSequencesLazily()
...
+                    case IEnumerable inner:
+                        return inner;
+                    default:
+                        return Enumerable.Empty<object>();
+                }
+            });
+
+            var expectations = Enumerable.Range(1, 7);
+
+            Assert.That(result, Is.EquivalentTo(expectations));
```
Comment (atifaziz) [inline MoreLinq.Test/FlattenTest.cs]:
```suggestion
            Assert.That(result, Is.EqualTo(expectations));
```

**Comment by leandromoh** [comment]
@atifaziz adjusts done!

**Comment by atifaziz** [comment]
Thank you!

**Pull request merged**